spring-data-relational/src/main/java/org/springframework/data/relational/core/conversion/MergedAggregateChange.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.data.relational.core.conversion;
+
+/**
+ * Represents the changes happening to one or more aggregates (as used in the context of Domain Driven Design) as a
+ * whole. This change allows additional {@link MutableAggregateChange} of a particular kind to be merged into it to
+ * broadly represent the changes to multiple aggregates across all such merged changes.
+ *
+ * @author Chirag Tailor
+ * @since 3.0
+ */
+public interface MergedAggregateChange<T, C extends MutableAggregateChange<T>> extends AggregateChange<T> {
```

Comment:
I'm not sure that extending `AggregateChange` is appropriate as `AggregateChange` defines a single underlying entity. I think we could extract an `AggregateChangeDefinition` (lacking a better name) interface defining `kind`/`entityType` and let `AggregateChange` extend `AggregateChangeDefinition`.

This type would also extend `AggregateChangeDefinition` and hold references to multiple `AggregateChange`s.

I would also suggest naming this type with a more clear name. `Merged change` (without further details initially) can mean that we work with a single aggregate where something else merges its changes or as defined here, that we handle multiple aggregates. How about `BatchAggregateChange` or `AggregateChangeBatch`?

Reply:
The behaviors from the `AggregateChange` hierarchy which indicated a change definition for a single entity were recently pulled out into a new [`AggregateChangeWithRoot` interface](https://github.com/spring-projects/spring-data-relational/blob/f2dc64ec52e361acdd5022032fb9dde27b9ca7bd/spring-data-relational/src/main/java/org/springframework/data/relational/core/conversion/AggregateChangeWithRoot.java), leaving `AggregateChange` to represent what you describe in `AggregateChangeDefinition`.

I agree that the naming of this type is not very clear. I will plan to update it to something that is hopefully better.

Reply:
Thanks for pointing that out. I think I was looking at the `2.3.x` branch where `AggregateChange` defined `T getEntity()` and that was what caused confusion on my side.

Also, `AggregateChangeWithRoot` has a `since 2.6` version to it that adds to more confusion.

After understanding the arrangement I agree with the approach as it is outlined here.

To resolve the naming issue, we could rename `AggregateChangeWithRoot` into `RootAggregateChange` which makes things much cleaner (also set the `@since` version to `3.0`).